editor/import/resource_importer_obj.cpp

```diff
@@ -220,8 +221,13 @@ static Error _parse_obj(const String &p_path, List<Ref<Mesh>> &r_meshes, bool p_
 	ERR_FAIL_COND_V_MSG(coff_header_machines.find(first_bytes) != -1, ERR_FILE_CORRUPT, vformat("Couldn't read OBJ file '%s', it seems to be binary, corrupted, or empty.", p_path));
 	f->seek(0);
 
-	Ref<ArrayMesh> mesh;
-	mesh.instantiate();
+	Ref<Resource> mesh;
+
+	if (p_use_array_mesh) {
+		mesh = Ref<ArrayMesh>(memnew(ArrayMesh));
```

Comment:
I feel there is not much need to use a direct to mesh version, you can still use ImporterMesh and convert to Mesh at the end, will not make any practical difference since ImporterMesh just keeps the arrays you parsed.

Reply:
Changed to use this structure.